### DISCUSSION TITLE
Installing the Nvidia datacenter drivers on AL2023

### DISCUSSION THREAD

**Discussion by openmarmot** [amazonlinux/amazon-linux-2022#649] [Show and tell]
[edit] - see the reply to this with the official (and much better) solution. 

In case anyone is struggling with the Nvidia driver installs for AL2023, I recently did the work to get the drivers working for the P3 series (Tesla V100)

```
          # install pre-reqs
          dnf install -y kernel-devel-$(uname -r) kernel-headers-$(uname -r) kernel-modules-extra
          
          # silent install of nvidia datacenter drivers for the p3 series
          BASE_URL=https://us.download.nvidia.com/tesla
          DRIVER_VERSION=550.54.14
          curl -fSsl -O $BASE_URL/$DRIVER_VERSION/NVIDIA-Linux-x86_64-$DRIVER_VERSION.run
          chmod +x NVIDIA-Linux-x86_64-$DRIVER_VERSION.run
          ./NVIDIA-Linux-x86_64-$DRIVER_VERSION.run --tmpdir . --silent
```

If you want a full cloudformation template - I have created some examples here : 
https://github.com/openmarmot/aws-ec2-nvidia-drivers

**Comment by limmike** [comment]
Thanks for sharing.
NVIDIA has added [repository support](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/#network-repo-installation-for-amazon-linux) for AL2023. This makes it convenient to update NVIDIA GPU drivers and CUDA toolkit.

Step-by-step instructions available at [re:Post](https://repost.aws/articles/ARwfQMxiC-QMOgWykD9mco1w/how-do-i-install-nvidia-gpu-driver-cuda-toolkit-and-optionally-nvidia-container-toolkit-in-amazon-linux-2023-al2023)

**Reply by openmarmot** [reply]
Awesome thanks! 
This is superior to what I was doing manually. I have tested and confirmed that it works on a p3.2xlarge

**Comment by elsaco** [comment]
There's a typo in the instructions at https://docs.nvidia.com/cuda/cuda-installation-guide-linux/#network-repo-installation-for-amazon-linux. Distro name should be `amzn2023` and not `amz2023`

Network install instructions from the online doc:

`sudo dnf config-manager --add-repo https://developer.download.nvidia.com/compute/cuda/repos/amz2023/x86_64/cuda-amz2023.repo
`
will result in a `404` error:

```
[ec2-user@amazon1 ~]$ sudo dnf config-manager --add-repo https://developer.download.nvidia.com/compute/cuda/repos/amz2023/x86_64/cuda-amz2023.repo
Adding repo from: https://developer.download.nvidia.com/compute/cuda/repos/amz2023/x86_64/cuda-amz2023.repo
Status code: 404 for https://developer.download.nvidia.com/compute/cuda/repos/amz2023/x86_64/cuda-amz2023.repo (IP: 152.195.19.142)
Error: Configuration of repo failed
```

**Reply by limmike** [reply]
The article at [Repost.aws](https://repost.aws/articles/ARwfQMxiC-QMOgWykD9mco1w/how-do-i-install-nvidia-gpu-driver-cuda-toolkit-and-optionally-nvidia-container-toolkit-in-amazon-linux-2023-al2023) uses the correct URI. 


As typo is on [their documentation site](https://docs.nvidia.com/cuda/index.html) , you can submit feedback to NVIDIA. The *Send feedback* link points to mailto:CUDAIssues@nvidia.com?subject=CUDA Toolkit Documentation Feedback. 

I have emailed NVIDIA about the typo.

**Reply by limmike** [reply]
NVIDIA has updated [AL2023 documentation](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/#network-repo-installation-for-amazon-linux) for CUDA 12.5 Update 1
